src/pages/dashboard/reset-pin.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box } from '@chakra-ui/react'
+import { useMutation } from '@tanstack/react-query'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+import CustomLoading from '@/components/CustomLoading'
+import ResetPinModal from '@/components/profile/components/ResetPinModal'
+import SuccessModal from '@/components/profile/components/SuccessModal'
+import { resetPinWithToken } from '@/utils/api_helper'
+import { handleApiError } from '@/utils/error_helper'
+import { useToastHelpers } from '@/utils/toasts'
+
+const ResetPinPage = () => {
+  const router = useRouter()
+  const { token, address } = router.query
```

Comment:
Why are we using the address from a query ? The user needs to be logged in to do this , yes ?